src/main/scala/mesosphere/marathon/MarathonSchedulerActor.scala

```diff
@@ -146,6 +146,8 @@ class MarathonSchedulerActor private (
     case cmd @ ScaleRunSpec(runSpecId) =>
       logger.debug("Receive scale run spec for {}", runSpecId)
 
+      // Acquiring the lock will prevent any scaling top happen if the app is already locked by an ongoing deployment.
+      // In this case, the deployment actor is expected to handle any instance state changes..
```

Comment:
Ideally we drop the scale altogether in a follow up.

Reply:
Totally agree